server/src/main/scala/scaladex/server/route/api/SearchApi.scala

```diff
@@ -252,4 +248,19 @@ class SearchApi(searchEngine: SearchEngine, database: WebDatabase, session: Gith
           project.githubInfo.flatMap(_.description).getOrElse("")
         )
       }
+
+  object AutocompletionApi extends AutocompletionEndpoints with server.Endpoints with server.JsonEntitiesFromSchemas {
+    // On the server-side, the session is made of a user id
+    case class WithSession[A](data: A, maybeUserId: Option[UUID])
+    def withOptionalSession[A](request: Request[A]): Request[WithSession[A]] =
+      new Request[WithSession[A]] {
+        val directive: Directive1[WithSession[A]] = Directive[Tuple1[WithSession[A]]] { k =>
+          optionalSession(refreshable, usingCookies) { maybeUserId =>
+            request.directive(a => k(Tuple1(WithSession(a, maybeUserId))))
+          }
```

Comment:
I manually tested (locally) that autocompleting “my projects” works as expected.